lang/syn/src/codegen/program/cpi.rs

```diff
@@ -70,11 +70,20 @@ pub fn generate(program: &Program) -> proc_macro2::TokenStream {
                 let sighash_arr = sighash(SIGHASH_GLOBAL_NAMESPACE, name);
                 let sighash_tts: proc_macro2::TokenStream =
                     format!("{:?}", sighash_arr).parse().unwrap();
+                let ret_type = &ix.returns.ty.to_token_stream();
+                let (method_ret, maybe_return) = match ret_type.to_string().as_str() {
+                    "()" => (quote! {anchor_lang::Result<()> }, quote! { Ok(()) }),
+                    _ => (
+                        quote! { anchor_lang::Result<Return::<#ret_type>> },
+                        quote! { Ok(Return::<#ret_type> { phantom: PhantomData }) }
```

Comment:
use full path for `Return` and `PhantomData` to avoid naming clashes (in the line above there's a `Return` too)